civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -315,50 +315,32 @@ function _rebuild_absence_activity_view() {
-                CONCAT(IF (DATE_FORMAT(a2.activity_date_time, '%w') = 0, 7, DATE_FORMAT(a2.activity_date_time, '%w')), '. ', DATE_FORMAT(a2.activity_date_time, '%W')) AS absence_day_of_week,
-                a2.duration AS absence_duration,
-                IF (ov.label <> 'TOIL (Credit)', a2.duration, NULL) AS absence_amount_taken,
-                IF (ov.label = 'TOIL (Credit)', a2.duration, NULL) AS absence_amount_accrued,
-                IF (ov.label = 'TOIL (Credit)', -a2.duration, a2.duration) AS absence_absolute_amount,
-                a.status_id AS absence_status,
-                IF (ov.label = 'TOIL (Credit)', 1, 0) AS absence_is_credit
-              FROM {$civi_db_name}.civicrm_activity absence_activity
-              LEFT JOIN {$civi_db_name}.civicrm_activity a ON a.id = absence_activity.id
-              LEFT JOIN {$civi_db_name}.civicrm_activity a2 ON a2.source_record_id = absence_activity.id
-              LEFT JOIN {$civi_db_name}.civicrm_option_group og ON og.name = 'activity_type'
-              LEFT JOIN {$civi_db_name}.civicrm_option_value ov ON ov.value = a.activity_type_id AND ov.option_group_id = og.id
-              LEFT JOIN {$civi_db_name}.civicrm_activity_contact ac ON ac.activity_id = absence_activity.id AND ac.record_type_id = 3
-              WHERE (a2.duration IS NOT NULL AND a2.duration > 0)
-              AND
-              absence_activity.activity_type_id IN (" . implode(',', $activityTypesIds) . ")");
+  db_query('DROP VIEW IF EXISTS absence_activity');
+  db_query("CREATE VIEW absence_activity AS
+            SELECT lr.contact_id AS absence_contact_id,
+              lr.id AS absence_activity_id,
+              at.title as absence_type,
+              lrd.date AS absence_date,
+              DATE_FORMAT(DATE(lrd.date), '%Y-%m') AS absence_month,
+              (SELECT MIN(DATE(date)) FROM {$civi_db_name}.civicrm_hrleaveandabsences_leave_request_date WHERE leave_request_id = lr.id) AS absence_start_date,
+              DATE_FORMAT((SELECT MIN(DATE(date)) FROM {$civi_db_name}.civicrm_hrleaveandabsences_leave_request_date WHERE leave_request_id = lr.id), '%Y-%m') AS absence_start_date_month,
+              (SELECT MAX(DATE(date)) FROM {$civi_db_name}.civicrm_hrleaveandabsences_leave_request_date WHERE leave_request_id = lr.id) AS absence_end_date,
+              DATE_FORMAT((SELECT MAX(DATE(date)) FROM {$civi_db_name}.civicrm_hrleaveandabsences_leave_request_date WHERE leave_request_id = lr.id), '%Y-%m') AS absence_end_date_month,
+              CONCAT(IF (DATE_FORMAT(lrd.date, '%w') = 0, 7, DATE_FORMAT(lrd.date, '%w')), '. ', DATE_FORMAT(lrd.date, '%W')) AS absence_day_of_week,
+              ABS(bc.amount) as absence_duration,
+              IF(lr.request_type != 'toil', ABS(bc.amount), NULL) AS absence_amount_taken,
+              IF(lr.request_type = 'toil', bc.amount, NULL) AS absence_amount_accrued,
+              IF(lr.request_type = 'toil', -bc.amount, -bc.amount) AS absence_absolute_amount,
```

Comment:
For some reason, On the old view, absolute amount for TOIL is always displayed in negative and positive for leave requests. 
To achieve this I had to negate toil since its always has positive balance changes on L&A and make that of leave requests positive since it always has negative balance changes.